src/manifest.c

```diff
@@ -481,6 +469,7 @@ static int try_delta_manifest_download(int current, int new, char *component, st
 	}
 
 	unlink(deltafile);
+	populate_file_struct(file, newfile);
 	ret = compute_hash(file, newfile); // MUST save new hash!
```

Comment:
Per testing and attempted implementation...this is not a feasible solution at this time, and we confirmed that calling populate_file_struct() resets the entire struct contents that could have been corrupt, so it is not needed to use a secondary struct.